Yoast/Sniffs/Commenting/CoversTagSniff.php

```diff
@@ -24,11 +24,9 @@ class CoversTagSniff implements Sniff {
 	/**
 	 * Regex to check for valid content of a @covers tags.
 	 *
-	 * Takes the WP naming conventions into account (up to a point).
-	 *
 	 * @var string
 	 */
-	const VALID_CONTENT_REGEX = '(?:\\\\?(?:(?<OOName>[A-Z][a-zA-Z0-9_\x7f-\xff]*)\\\\)*(?P>OOName)(?:<extended>|::<[!]?(?:public|protected|private)>|::(?<functionName>(?!public$|protected$|private$)[a-z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*))?|::(?P>functionName)|\\\\?(?:(?P>OOName)\\\\)+(?P>functionName))';
+	const VALID_CONTENT_REGEX = '(?:\\\\?(?:(?<OOName>[a-zA-Z_\x80-\xff][a-zA-Z0-9_\x80-\xff]*)\\\\)*(?P>OOName)(?:<extended>|::<[!]?(?:public|protected|private)>|::(?<functionName>(?!public$|protected$|private$)(?P>OOName)))?|::(?P>functionName)|\\\\?(?:(?P>OOName)\\\\)+(?P>functionName))';
```

Comment:
Could you explain what this does:

`[a-zA-Z_\x80-\xff][a-zA-Z0-9_\x80-\xff]*` versus `[A-Z][a-zA-Z0-9_\x7f-\xff]*`

The internet tells me it's extending the range of allowed characters beyond ASCII; is this for multibyte language support or am I thinking too far out?

Reply:
This is the official regex as per the PHP documentation for what is allowed as an object/function name.
It will now allow for all PHP supported names, not just the ones allowed according to the WP coding standards.
 
Example which is perfectly valid PHP (but please don't use):
```php
class 💩💩💩💩
{
	function 💩💩💩($😎, $🐯)
	{
		return $😎 + $🐯;
	}

}
$🐔 = 3;
$😥 = $🐔 + 2;
$💩 = new 💩💩💩💩;
echo $💩 -> 💩💩💩($🐔, $😥);
```

Refs;
* https://www.php.net/manual/en/language.oop5.basic.php
* https://www.php.net/manual/en/functions.user-defined.php

Reply:
Thanks, that is just so simple, and also very terrifying (the example).